src/System.Buffers/src/System/Buffers/Binary/BinaryHelpers.cs

```diff
@@ -0,0 +1,43 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Reflection;
+
+namespace System.Buffers.Binary
+{
+    internal static class BinaryHelpers
+    {
+        /// <summary>
+        /// Determine if a type is eligible for storage in unmanaged memory.
+        /// Portable equivalent of RuntimeHelpers.IsReferenceOrContainsReferences&lt;T&gt;()
+        /// </summary>
+        public static bool IsReferenceOrContainsReferences<T>() => IsReferenceOrContainsReferencesCore(typeof(T));
```

Comment:
I believe this should actually read as follows for best performance.

```csharp
private static class IsReferenceImpl<T> {
  public static bool IsReferenceOrContainsReferences = IsReferenceOrContainsReferenceCore(typeof(T));
}

public static bool IsReferenceOrContainsReferences<T>() => IsReferenceImpl<T>.IsReferenceOrContainsReferences;
```

Moving the `<T>` onto a class rather than in a method allows the JITter to cache the value per T.

Reply:
Right, it should be done exactly like it is done for Span: https://github.com/dotnet/corefx/blob/master/src/System.Memory/src/System/SpanHelpers.cs#L129

If we end up compiling in multiple copies, the implementation should be moved under Common.

Reply:
Once we have the blittable constraint, this code will no longer be necessary here. Should we still move it to Common?

Reply:
You do not need to move it now that we have decided to add these APIs to System.Memory.